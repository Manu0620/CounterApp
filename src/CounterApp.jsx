import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }) => {
   
    const [ count, setCount ] = useState( value ); 

    const handleAdd = () => setCount( count + 2 );

    const handleSubtract = () => {
        if( count === 0 )
            return;
        setCount( count - 2);
    }

    const handleReset = () => setCount( value );

    return(
        <>
            <h1>Counter App</h1>
            <h3>Count: { count }</h3>

            <button onClick={ handleSubtract }> -2 </button>
            <button onClick={ handleAdd }> +2 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> ‚ÜêReset </button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 'No hay valor!'
}