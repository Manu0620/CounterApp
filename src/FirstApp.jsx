import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {
    
    //console.log(props);

    return (
        <>
            <h1>{ title }</h1>
            {/* <code>{ JSON.stringify( newMessage ) }</code> */}
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Manuel Madera',
    title: 'No hay titulo!',
    subtitle: 'No hay subtitulo!'
}