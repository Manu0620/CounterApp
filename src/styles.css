*{
    font-family: Space Mono, Arial, sans-serif;
}

html, body{
    display: flex;
    justify-content: center;
    text-align: center;
    background-color: #21232A;
    color: lightblue;
    font-size: 1rem;
    padding: 60px;
}

button{
    background-color: lightblue;
    border: 2px solid #21232A;
    border-radius: 10px;
    color: #21232A;
    font-weight: bold;
    padding: 10px;
    margin: 5px;
}

button:hover{
    background: rgba(173, 216, 230, 0.674);
    color: #2a2c36;
}