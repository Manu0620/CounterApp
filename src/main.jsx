import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
// import { HelloWorldApp } from './HelloWorldApp';
//import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Hola, Manuel' subtitle='Te saluda:' name='Fernando Herrera' /> */}
        <CounterApp value={ 10 } />
    </React.StrictMode>
)