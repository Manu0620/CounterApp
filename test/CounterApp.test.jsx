import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Pruebas en <CounterApp />', () => {

    const value = 10;

    test('debe de hacer match con el snapshot', () => {

        const { container } = render(  
            <CounterApp 
                value={ value }
            />           
        )

        expect( container ).toMatchSnapshot();

    })

    test('debe de mostar el valor inicial en 10', () => {

        render(
            <CounterApp
                value={ value }
            />
        )

        expect( screen.getByRole( 'heading', { level: 3 }).innerHTML ).toBe( `Count: ${ value }` );

    })

    test('debe de incrementar con el boton +2', () => {

        render(     
            <CounterApp 
                value={ value }
            />
        );

        fireEvent.click( screen.getByText('+2') )
        expect( screen.getByText(`Count: 12`) ).toBeTruthy();

    })

    test('debe de decrementar con el boton -2', () => {

        render(
            <CounterApp
                value={value}
            />
        );

        fireEvent.click(screen.getByText('-2'))
        expect(screen.getByText(`Count: 8`)).toBeTruthy();

    })

    test('debe de funcionar el boton de reset', () => {

        render(
            <CounterApp
                value={value}
            />
        );
        fireEvent.click(screen.getByText('+2'))
        fireEvent.click(screen.getByText('+2'))
        fireEvent.click(screen.getByText('+2'))
        //fireEvent.click(screen.getByText('‚ÜêReset'))
        fireEvent.click(screen.getByRole( 'button', { name: 'btn-reset' } ));

        expect( screen.getByText(`Count: ${ value }`) ).toBeTruthy();

    })

})