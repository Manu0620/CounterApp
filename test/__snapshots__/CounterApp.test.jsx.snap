// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h3>
    Count: 
    10
  </h3>
  <button>
     -2 
  </button>
  <button>
     +2 
  </button>
  <button
    aria-label="btn-reset"
  >
     ‚ÜêReset 
  </button>
</div>
`;
